packages/darklang/cli/integrationTests/test-framework.dark

```diff
@@ -0,0 +1,283 @@
+          { name: String
+            steps: List<InteractiveStep>
+            timeoutSeconds: Int64 }
+
+        type CliTest =
+          { name: String
+            arguments: List<String>
+            expectedOutputFile: String
+            isInteractive: Bool
+            interactiveTest: Stdlib.Option.Option<InteractiveCliTest> }
+
+        type TestResult =
+          | Success of message: String
+          | Failure of message: String
+          | SetupComplete of message: String
+
+
+        let makeCliTest
+          (name: String)
+          (arguments: List<String>)
+          (expectedOutputFile: String): CliTest =
+          CliTest
+            { name = name
+              arguments = arguments
+              expectedOutputFile = expectedOutputFile
+              isInteractive = false
+              interactiveTest = Stdlib.Option.Option.None }
+
+        let makeInteractiveCliTest
+          (name: String)
+          (steps: List<InteractiveStep>)
+          (timeoutSeconds: Int64): CliTest =
+          let interactiveTest =
+            InteractiveCliTest
+              { name = name
+                steps = steps
+                timeoutSeconds = timeoutSeconds }
+          CliTest
+            { name = name
+              arguments = []
+              expectedOutputFile = ""
+              isInteractive = true
+              interactiveTest = Stdlib.Option.Option.Some interactiveTest }
+
+
+        let createDirectories (): Stdlib.Result.Result<Unit, String> =
+          match TestUtils.createDirectory (TestConfig.getActualDir ()) with
+          | Error e -> Stdlib.Result.Result.Error $"Failed to create actual directory: {e}"
+          | Ok _ ->
+            match TestUtils.createDirectory (TestConfig.getExpectedDir ()) with
+            | Error e -> Stdlib.Result.Result.Error $"Failed to create expected directory: {e}"
+            | Ok _ -> Stdlib.Result.Result.Ok ()
+
+
+        let runCliCommand (command: String) (outputFile: String): Stdlib.Result.Result<Unit, String> =
+          let fullCommand = $"cd {TestConfig.projectRoot} && {command} > {outputFile} 2>&1"
+          let result = Builtin.cliExecute fullCommand
+          if result.exitCode == 0L then
+            Stdlib.Result.Result.Ok ()
+          else
+            Error $"CLI command failed with exit code {Stdlib.Int64.toString result.exitCode}: {result.stderr}"
+
+
+        let isFileEmpty (filePath: String): Bool =
+          match TestUtils.getFileContent filePath with
+          | Ok content -> Stdlib.Bool.not (Stdlib.String.isEmpty content)
+          | Error _ -> false
+
+        let filesAreEqual (file1: String) (file2: String): Bool =
+          let result = Builtin.cliExecute $"diff -q {file1} {file2}"
+          result.exitCode == 0L
+
+        let generateDiff (expectedFile: String) (actualFile: String): String =
+          let result = Builtin.cliExecute $"diff --unified=3 --color=always {expectedFile} {actualFile}"
```

Comment:
won't work on Windows. We're not on Windows, so that's fine for now, but we should be careful in a few ways:
- if something only works on UNIX systems, note so (e.g. in the `integrationTests/README.md?`)
- generally avoid using `Builtin.cliExecute` in code outside of stdlib -- and use it as little as possible.
  - to that end, this may be a good time to add a `Darklang.Stdlib.CLI.diff` -- which (for now) can work on Unix doing what you did, and fail/RTE on Windows

Reply:
rethinking, some of this feedback doesn't make sense since we'll likely always run the tests in the devcontainer, but it's probably a good habit to get into?